File: lua/refactoring/tests/refactor/119/c/example/extract_var.expected.c
#include <stdio.h>

#define max(a, b)                                                              \
  ({                                                                           \
    __typeof__(a) _a = (a);                                                    \
    __typeof__(b) _b = (b);                                                    \
    _a > _b ? _a : _b;                                                         \
  })

#define min(a, b)                                                              \
  ({                                                                           \
    __typeof__(a) _a = (a);                                                    \
    __typeof__(b) _b = (b);                                                    \
    _a < _b ? _a : _b;                                                         \
  })

struct Order {
  double quantity;
  double itemPrice;
};

double orderCalculation(struct Order order, int a, int b) {
  float temp = a * b;
  INSERT_TYPE_HERE basePrice = order.quantity * order.itemPrice;
  float blah = (basePrice) - 7;

  printf("%f %f", temp, blah);

  return basePrice -
         max(0.0, order.quantity - 500) * order.itemPrice * 0.05 +
         min(basePrice * 0.1, 100.00);
}

File: lua/refactoring/tests/refactor/119/ts/example/extract_var.expected.ts
type Order = {
    quantity: number;
    itemPrice: number;
}

// This the straight from the book (slight modifications)
order.quantity * order.itemPrice;
function orderCalculation(order: Order) {
    foo.blah * nonsense.nonsense;
    const basePrice = order.quantity * order.itemPrice;
    basePrice;
    return basePrice -
        Math.max(0, order.quantity - 500) * order.itemPrice * 0.05 +
        Math.min(basePrice * 0.1, 100);
}
